packages/typespec-autorest/test/paging.test.ts

```diff
@@ -38,6 +38,28 @@ it("define a custom paged operation with custom next link", async () => {
   });
 });
 
+it("define a custom paged operation with nested next link and items", async () => {
+  const res = await openApiFor(
+    `
+    model CustomPageModel<T> {
+      results: { @pageItems myItems: T[] };
+      pagination: {
+        @TypeSpec.nextLink
+        \`@odata.nextLink\`?: string;
+      };
+    }
+    @list op list(): CustomPageModel<{}>;
+    `,
+  );
+
+  const listThings = res.paths["/"].get;
+  ok(listThings);
+  deepStrictEqual(listThings["x-ms-pageable"], {
+    nextLinkName: "pagination.@odata.nextLink",
```

Comment:
is that actually the correct path if the property contains a `.`? How does autorest differentiate

Reply:
@tadelesh Can you comment on the emitter '.' Handling behavior?

Reply:
tcgc export the path with property array to prevent `.` ambiguous in string path.

Reply:
@tadelesh, it is unclear that `x-ms-pageable` supports nested property references AND TypeSpec autorest emitter had code explicitly take the last segment to output to the `x-ms-pageable` fields, I will close this PR and be done with the original issue. The full path is available on `PageProperty.path` if DPG/MPG decides to make use of it.

Closing this PR.